include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -282,15 +301,15 @@ def AndOfPad : Pat <
   (MoveNameHint $old, (PadPrimOp (AndPrimOp (TailPrimOp $x, (TypeWidthAdjust32 $x, $y)), $y), $n)),
   [(KnownWidth $x), (UIntType $x), (EqualTypes $x, $pad)]>;
 
-/// cvt is adding a 0, which will pass through the and
-/// and(cvt(x:U), const) -> cat(0, and(x,trunc(const))
-def AndCvtU : Pat <
-  (AndPrimOp:$old (CvtPrimOp $x), (ConstantOp:$ocst $cst)),
-  (MoveNameHint $old, (CatPrimOp 
-    (NativeCodeCall<"$_builder.create<ConstantOp>($0.getLoc(), APSInt(APInt()))"> $ocst),
-    (AndPrimOp $x, (NativeCodeCall<"$_builder.create<ConstantOp>($0.getLoc(), cast<IntType>($0.getType()), $1.getValue().trunc($1.getValue().getBitWidth()-1).zext(*cast<IntType>($0.getType()).getWidth()))"> $x, $cst))
-  )),
-  [(KnownWidth $x), (UIntType $x)]>;
+def AndOfAsSIntL : Pat<
+  (AndPrimOp:$old (AsSIntPrimOp $x), $y),
+  (MoveNameHint $old, (AndPrimOp $x, (AsUIntPrimOp $y))),
+  []>;
+
+def AndOfAsSIntR : Pat<
+  (AndPrimOp:$old $x, (AsSIntPrimOp $y)),
+  (MoveNameHint $old, (AndPrimOp (AsUIntPrimOp $x), $y)),
+  []>;
```

Comment:
This is incorrect when `sizeof(x) != sizeof(y)` since implicit extension needs to happen. 
```suggestion
def AndOfAsSIntL : Pat<
  (AndPrimOp:$old (AsSIntPrimOp $x), $y),
  (MoveNameHint $old, (AndPrimOp $x, (AsUIntPrimOp $y))),
  [(KnownWidth $x), (EqualIntSize $x, $y)]>;

def AndOfAsSIntR : Pat<
  (AndPrimOp:$old $x, (AsSIntPrimOp $y)),
  (MoveNameHint $old, (AndPrimOp (AsUIntPrimOp $x), $y)),
  [(KnownWidth $x), (EqualIntSize $x, $y)]>;
```